File: frontend/src/app/model/entities/work-hours.ts
import {BaseMoikaEntity} from "./base-moika-entity";
import {LocalTime} from "./local-time";

export class WorkHours extends BaseMoikaEntity {
  timeOnStarts: LocalTime;
  timeOnEnds: LocalTime;

  constructor(ts: LocalTime, te: LocalTime) {
    super();
    this.timeOnStarts = ts;
    this.timeOnEnds = te;
  }

  public static fromArray(ts: number[], te: number[]) {
    let _tsA: number[] = [0, 0, 0, 0];
    let _teA: number[] = [0, 0, 0, 0];
    let _al: number;
    _al = (ts.length >= 4) ? 4 : ts.length;
    for (var i = 0; i < _al; i++) _tsA[i] = ts[i];

    _al = (te.length >= 4) ? 4 : te.length;
    for (var i = 0; i < _al; i++) _teA[i] = te[i];

    let _ts = new LocalTime(ts[0], ts[1], ts[2], ts[3]);
    let _te = new LocalTime(te[0], te[1], te[2], te[3]);
    return new WorkHours(_ts, _te);
  }

  public static parse(ts: string, te: string): WorkHours {
    let _ts = LocalTime.parse(ts);
    let _te = LocalTime.parse(te);
    return new WorkHours(_ts, _te);
  }

  public toDate(): Array<Date>{
    let _retA = new Array<Date>();
    _retA.push(this.timeOnStarts.toDate());
    _retA.push(this.timeOnEnds.toDate())
    return _retA;
  }
}

File: backend/src/main/java/io/khasang/moika/entity/WorkHours.java
package io.khasang.moika.entity;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.Id;
import java.math.BigDecimal;
import java.time.LocalTime;
import java.time.format.DateTimeParseException;
import java.util.Objects;

/**
 * Сущность, описывающая рабочие часы (начало,окончание)
 */
@Embeddable
public class WorkHours {
    private static final Logger logger = LoggerFactory.getLogger(WorkHours.class);
    @Id
    @Column(name = "time_on_starts")
    protected LocalTime timeOnStarts = LocalTime.of(0, 0);

    @Column(name = "time_on_ends")
    protected LocalTime timeOnEnds = LocalTime.of(0, 0);

    public WorkHours() {
    }

    public WorkHours(LocalTime start, LocalTime end) {
        this.timeOnStarts = start;
        this.timeOnEnds = end;
    }


    public WorkHours(String start, String end) {
        try {
            this.timeOnStarts = LocalTime.parse(start);
            this.timeOnEnds = LocalTime.parse(end);
        } catch (DateTimeParseException e) {
            logger.error("Incorrect local time initialization!!! start:" + start + " : " + "end:" + end);
        }
    }

    public LocalTime getTimeOnStarts() {
        return timeOnStarts;
    }

    public void setTimeOnStarts(LocalTime timeOnStarts) {
        this.timeOnStarts = timeOnStarts;
    }

    public LocalTime getTimeOnEnds() {
        return timeOnEnds;
    }

    public void setTimeOnEnds(LocalTime timeOnEnds) {
        this.timeOnEnds = timeOnEnds;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof WorkHours)) return false;
        WorkHours workHours = (WorkHours) o;
        return Objects.equals(timeOnStarts, workHours.timeOnStarts) &&
                Objects.equals(timeOnEnds, workHours.timeOnEnds);
    }

    @Override
    public int hashCode() {
        return Objects.hash(timeOnStarts, timeOnEnds);
    }
}
